app/src/main/java/org/simple/clinic/bp/entry/BloodPressureEntrySheetControllerV2.kt

```diff
@@ -449,7 +449,7 @@ class BloodPressureEntrySheetControllerV2 @Inject constructor(
         .withLatestFrom(newBpDataStream) { _, newBp -> newBp }
         .ofType<NewBpData.ReadyToSave>()
         .flatMapSingle { (date, bp, patientUuid) ->
-          val dateAsInstant = date.atStartOfDay(UTC).toInstant()
+          val dateAsInstant = date.atTime(OffsetTime.now(clock)).toInstant()
```

Comment:
Just realized that I cannot use the injected clock here. Need to inject `ZoneId` separately.

Reply:
We're using the clock in a lot of places where we'll need to inject the zone separately.

Reply:
Doing this

Reply:
On a second thought, this is correct, right? We don't need to care about the user's time zone. The user's timezone is only needed when we need to display the time.

Reply:
Yes, you're correct. This specific instance needs the UTC time.